core/src/Cache/MsalRefreshTokenCacheItem.cs

```diff
@@ -25,68 +25,57 @@
 //
 //------------------------------------------------------------------------------
 
-using System.Globalization;
-using System.Runtime.Serialization;
 using Microsoft.Identity.Client.Internal;
 using Microsoft.Identity.Core.OAuth2;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
 
 namespace Microsoft.Identity.Core.Cache
 {
     [DataContract]
-    internal class MsalRefreshTokenCacheItem : MsalTokenCacheItemBase
+    internal class MsalRefreshTokenCacheItem : MsalCredentialCacheItemBase
     {
-
-        public MsalRefreshTokenCacheItem()
+        internal MsalRefreshTokenCacheItem()
         {
+            CredentialType = Cache.CredentialType.RefreshToken.ToString();
         }
-
-        public MsalRefreshTokenCacheItem(string environment, string clientId, MsalTokenResponse response) : base(clientId)
+        internal MsalRefreshTokenCacheItem(string environment, string tenantId, string clientId, MsalTokenResponse response) : this()
```

Comment:
sorry did not get the question, please explain more

Reply:
I believe I saw additional constructors earlier.. must have been dreaming. I wanted you to call other contructors using this instead of just the base constructor.

---
In reply to: [191035701](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1065#discussion_r191035701) [](ancestors = 191035701)